keycloak/realm.go

```diff
@@ -92,6 +110,14 @@ func (keycloakClient *KeycloakClient) ValidateRealm(realm *Realm) error {
 		return err
 	}
 
+	if (SmtpServer{}) != realm.SmtpServer && realm.SmtpServer.Host == "" {
```

Comment:
is there a reason why this is not checked by Terraform?
This attribute is configured as `Optional: true,`
=> could it be configured as`Required: true,`?

Reply:
Well it is an optional field in the sense that you do need to set smtp server, but the moment you set one field of the smtp server, this is one of two required fields. (host and from).
For that reason, I do not think I can make the field required, can I?
Not really a Golang orTerraform expert, But I will take a look tomorrow, what it gives when I make this field required but not set the smtpserver

Reply:
changed it

Reply:
thanks for reviewing this @BernhardBerbuir!  I would have left the same comment :smile: